backends-velox/src/test/scala/org/apache/spark/sql/execution/VeloxParquetWriteForHiveSuite.scala

```diff
@@ -330,4 +330,43 @@ class VeloxParquetWriteForHiveSuite
       }
     }
   }
+
+  testWithSpecifiedSparkVersion(
+    "Native writer should keep the same compression codec if `hive.exec.compress.output` is true",
+    maxSparkVersion = Some("3.3")) {
```

Comment:
It seems no need to test with specified spark version due to spark 3.4 and later don't support native write for ctas [here](https://github.com/apache/incubator-gluten/blob/main/gluten-substrait/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala#L92)? If so, spark 3.4 and later will fallback to vanilla spark and align with vanilla spark behavior.

Reply:
ctas will go now v2 write code path, will use `WriteFilesExecTransformer`. 

`WriteFilesExecTransformer` also has the problem described in this PR, but it seems more trickier, so this PR first fixes the v1 write path.